ordersCostumes.js

```diff
@@ -0,0 +1,117 @@
+const addCostumeToOrder = async (pool, costumeId, orderId) => {
+    // throw error if costume id does not exist
+    const {rows:[costume]} = await pool.query(`
+        SELECT * FROM costumes
+        WHERE id = $1;
+    `, [costumeId])
+    if (costume === undefined) {
+        throw new Error(`Could not retrieve data because id provided for costume (${id}) does not exist in table.`)
+    } 
+
+    // throw error if order id does not exist
+    const {rows:[order]} = await pool.query(`
+        SELECT * FROM orders
+        WHERE id = $1;
+    `, [orderId])
+    if (order === undefined) {
+        throw new Error(`Could not retrieve data because id provided for order (${id}) does not exist in table.`)
+    } 
+    
+    // add costume
+    const {rows: [costumeOrder]} = await pool.query(`
+        INSERT INTO orders_costumes(
+            costume_id,
+            order_id
+        ) VALUES ($1,$2)
+        RETURNING *;
+    `, [costumeId, orderId]
+    )
+    return costumeOrder;
+}
+
+const removeCostumeFromOrder = async (pool, costumeId, orderId) => {
+    // throw error if costume id does not exist
+    const {rows:[costume]} = await pool.query(`
+        SELECT * FROM costumes
+        WHERE id = $1;
+    `, [costumeId])
+    if (costume === undefined) {
+        throw new Error(`Could not retrieve data because id provided for costume (${id}) does not exist in table.`)
+    } 
+
+    // throw error if order id does not exist
+    const {rows:[order]} = await pool.query(`
+        SELECT * FROM orders
+        WHERE id = $1;
+    `, [orderId])
+    if (order === undefined) {
+        throw new Error(`Could not retrieve data because id provided for order (${id}) does not exist in table.`)
+    } 
```

Comment:
Same, this shouldn't be necessary, won't SQL error if you pass invalid IDs or no IDs?

Reply:
Also this is repeat code - it would be better to check this kind of thing not at the DB adapter layer (eg. in the web server, you might re-use getOrder and getCostume first to check)

Reply:
OK. I'll remove it for now.